### PR TITLE
Resolving Line URIs  

### PR DESCRIPTION
**Pull request by DrewSadler01**

### CODE REVIEW

File: src/java/edu/slu/tpen/servlet/LineServlet.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+     *
+     * @param req servlet request
+     * @param resp servlet response
+     * @throws ServletException if a servlet-specific error occurs
+     * @throws IOException if an I/O error occurs
+     */
+    
+    //A callable method that would have the framework to resolve URI and a header request to the page
+    @Override
+    protected void doGetUri(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
```
Comment (thehabes) [inline src/java/edu/slu/tpen/servlet/LineServlet.java:47]: I think you are looking for [HTTPServlet's doGet](https://docs.oracle.com/javaee/7/api/javax/servlet/http/HttpServlet.html) here,  which you intend to `@override`

File: src/java/edu/slu/tpen/servlet/LineServlet.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    protected void doPut(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+       doGet(req, resp);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getServletInfo() {
+        return "T-PEN Canvas Dereferencer";
```
Comment (thehabes) [inline src/java/edu/slu/tpen/servlet/LineServlet.java:107]: Line not Canvas

**Review by thehabes — CHANGES_REQUESTED** [review]
You have the boilerplate code.  You need to make it Line specific and forget about the Folios and Canvases.

File: src/java/edu/slu/tpen/servlet/LineServlet.java [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+                JSONObject Writer = new JSONObject();
+                Writer.accumulate("hello", "does this work");
+                resp.setContentType("application/json; charset=UTF-8");
+                resp.setHeader("Access-Control-Allow-Headers", "*");
+                resp.setHeader("Access-Control-Expose-Headers", "*"); //Headers are restricted, unless you explicitly expose them.  Darn Browsers.
+                resp.setHeader("Cache-Control","no-cache, no-store, must-revalidate,max-age=15"); // HTTP 1.1.");
+                resp.setHeader("Pragma", "no-cache"); // HTTP 1.0.
+                resp.setHeader("Expires", "0"); // Proxies.
+                resp.getWriter().write((Writer.toString()));
+                  /*
```
Comment (cubap) [inline src/java/edu/slu/tpen/servlet/LineServlet.java:68]: too much comment. We've got versioning; just make it disappear.

**Pull request closed without merging**